src/plugins/unified_search/public/search_bar/create_search_bar.tsx

```diff
@@ -212,7 +212,9 @@ export function createSearchBar({
             showFilterBar={props.showFilterBar}
             showQueryMenu={props.showQueryMenu}
             showQueryInput={props.showQueryInput}
-            showSaveQuery={props.showSaveQuery}
+            showSaveQuery={
+              Boolean(core.application.capabilities.globalSavedQueries?.edit) || props.showSaveQuery
+            }
```

Comment:
Maybe rather as a fallback?

```suggestion
            showSaveQuery={props.showSaveQuery ?? Boolean(core.application.capabilities.globalSavedQueries?.edit)}
```

Reply:
But with the following code in Discover, you want to disable it for text-based searches, no?
```
const showSaveQuery =
    !isPlainRecord &&
    (Boolean(services.capabilities.discover.saveQuery) ||
      Boolean(services.capabilities.globalSavedQueries.edit));
```

Reply:
You're right! Great catch! I guess we would need to solve that case in search_bar.tsx, disabling save queries for `ES|QL` or other text based langs. This makes also more sense than just solving it in a single plugin 👍